AEMG/data_utils.py

```diff
@@ -95,12 +95,72 @@ def __init__(self,config):
         self.labels = np.array(self.labels)
         self.labels = torch.from_numpy(self.labels).long()
 
+        self.generate_contrastive_triplets()
+
     def __len__(self):
-        return len(self.final_points)
+        return len(self.contrastive_triplets)
 
     def __getitem__(self, idx):
-        return self.final_points[idx], self.labels[idx]         
-            
+        return self.contrastive_triplets[idx]
+
+    def generate_contrastive_triplets(self):
+        """
+        Generates the contrastive triplets for the dataset where each triplet has the form (anchor, positive, negative)
+        The anchor can be either a success or failure point, the positive is a point of the same class as the anchor, and the negative is a point of the opposite class
+        """
+
+        # Gathering the indices of the success and failure labels
+        success_indices = torch.where(self.labels == 1)[0]
+        failure_indices = torch.where(self.labels == 0)[0]
+
+        # Generating all possible combinations of success and failure indices for positive anchor pairs
+        success_anchor_pairs = torch.stack(torch.meshgrid(success_indices, success_indices)).T.reshape(-1,2)
```

Comment:
This `reshape` (and following operations) should be to `(-1, config['low_dims']).`

Reply:
I don't think so. The two here correspond to the pairs and not really the dimensionality of the space. We're only working with indices here